tensorflow/compiler/jit/kernels/xla_ops.cc

```diff
@@ -1042,11 +1042,22 @@ REGISTER_KERNEL_BUILDER(Name("_XlaCompile")
                             .HostMemory("resources"),
                         XlaCompileOp);
 
+REGISTER_KERNEL_BUILDER(Name("_XlaCompile")
+                            .Device(DEVICE_DEFAULT)
```

Comment:
Register for "DEVICE_DEFAULT" is for the fallback kernels?

Also to double check, do you think there is any other side effect for registering for "DEVICE_DEFAULT"?

Reply:
@jyingl3 yes, for those third_party device expect CPU/GPU/TPU, they will use DEVICE_DEFAULT kernels. I think it  it would be ok since these kernels are for auto clustering mode, for those NextPluggableDevice who don't register XLA_GPU_JIT as jit_device_type, it will failed to get the compilation device and won't rewrite the graph to _XlaCompile and _XlaRun

Reply:
Got it. Thanks for the clarification!